### PR TITLE
Set up go function pointer table

### PR DESCRIPTION
**Pull request by wheatman**
To use usys in go we cause a GP fault which is then caught with known
high 16 bits.

The lower 48 bits are used to specify the address of a jump table which
go will use for usys function calls

Change-Id: Id0698f04b784a9d3376c10dd8fcbaebf8633cf50
Signed-off-by: Brian Wheatman <wheatman@google.com>

### CODE REVIEW

File: user/pthread/pthread.c [hunk 1]
```diff
@@ -21,6 +21,7 @@
 #include <parlib/stdio.h>
 #include <sys/fork_cb.h>
 
+
```
Comment (brho) [inline user/pthread/pthread.c]: can remove this extra newline

File: user/pthread/pthread.c [hunk 2]
```diff
@@ -304,9 +305,88 @@ static void handle_div_by_zero(struct uthread *uthread, unsigned int err,
 	__signal_and_restart(uthread, SIGFPE, FPE_INTDIV, (void*)aux);
 }
 
+#include <parlib/alarm.h>
+#include <futex.h>
+#include <parlib/serialize.h>
```
Comment (brho) [inline user/pthread/pthread.c]: can move these to the top of the file, under the other includes.

File: user/pthread/pthread.c [hunk 3]
```diff
@@ -304,9 +305,88 @@ static void handle_div_by_zero(struct uthread *uthread, unsigned int err,
...
+
+// checks that usys in go passes its arguments correctly
+// it only automatically checks with 7 arguments, print is for the rest
+int go_usys_tester(uint64_t a, uint64_t b, uint64_t c, uint64_t d, uint64_t e,
+                   uint64_t f, uint64_t g, uint64_t h, uint64_t i, uint64_t j,
+                   uint64_t k, uint64_t l)
+{
+        printf("a = %lu, b = %lu, c = %lu, d = %lu, e = %lu, f = %lu, g = %lu, h = %lu, i = %lu, j = %lu, k = %lu, l = %lu\n",
```
Comment (brho) [inline user/pthread/pthread.c]: these spaces should be a tab.  same with the next line down

**Comment by wheatman** [comment]
I made all the changes

**Comment by brho** [comment]
Merged to master at e94ea40da19b..09363dc1775f (from, to]

You can see the entire diff with 'git diff' or at
https://github.com/brho/akaros/compare/e94ea40da19b...09363dc1775f

**Pull request closed without merging**